replication/v1/replication.proto

```diff
@@ -162,3 +181,39 @@ message DeploymentReplication {
     // Status of the DeploymentReplication.
     Status status = 100;
 }
+
+// DeploymentMigration defines a request for performing the migration of a deployment.
+message DeploymentMigration {
+    // Identifier of the source deployment that needs to be migrated.
+    // This is a required field.
+    string source_deployment_id = 1;
+    // Specification of the target deployment.
+    data.v1.Deployment target_deployment = 2;
```

Comment:
Maybe in the public API instead of a complete data.deployment we should add a type covering the used properties only. I see in the implementation only a few are copied / used. This can lead to situations that someone updates more and those settings are lost.

Reply:
Updated